pkg/appctx/container.go

```diff
@@ -34,26 +34,26 @@ func WithContainer(ctx context.Context) context.Context {
 	})
 }
 
-func Provide(ctx context.Context, key interface{}, factory func() (interface{}, error)) (interface{}, error) {
+func Provide[T any](ctx context.Context, key any, factory func() (T, error)) (T, error) {
 	var ok bool
 	var err error
 	var contI, val interface{}
 
 	if contI = ctx.Value(containerKey); contI == nil {
-		return nil, &ErrNoApplicationContainerFound{}
+		return *new(T), &ErrNoApplicationContainerFound{}
```

Comment:
Why returning some empty type here instead of continuing returning nil?  
Usually we always return nil on error..